native/packages/react-native-bpk-component-button/src/BpkButton.js

```diff
@@ -64,13 +66,35 @@
    return styleForElement;
  };
 
- const getGradientColors = ({ type, disabled, selected }) => {
-   let gradientColors = styles.gradientColors[type];
+ const getThemingForElement = (elementType, theme, { type, disabled, selected }) => {
+   const themeForElement = {};
+   if (theme && styles.themeMappings[elementType]) {
+     Object.keys(styles.themeMappings[elementType]).forEach((key) => {
+       const values = styles.themeMappings[elementType][key];
+
+       if (disabled && theme[values.disabled]) {
+         themeForElement[key] = theme[values.disabled];
+       } else if (selected && theme[values.selected]) {
+         themeForElement[key] = theme[values.selected];
+       } else if (values[type]) {
+         themeForElement[key] = theme[values[type]];
+       }
+     });
+   }
+   return themeForElement;
+ };
+
+ const getGradientColors = (theme, { type, disabled, selected }) => {
+   let gradientColors = theme ?
+   [theme.gradientStartColor, theme.gradientEndColor] : styles.gradientColors[type];
+
    if (selected) {
-     gradientColors = styles.gradientColors.selected;
+     gradientColors = theme ?
+     [theme.selectedGradientStartColor, theme.selectedGradientEndColor] : styles.gradientColors.selected;
    }
    if (disabled) {
-     gradientColors = styles.gradientColors.disabled;
+     gradientColors = theme ?
+     [theme.disabledBackgroundColor, theme.disabledBackgroundColor] : styles.gradientColors.disabled;
    }
    return gradientColors;
```

Comment:
Once we support more types this will need to become more complex, but for now it's sufficient.